app/src/org/commcare/models/AndroidSessionWrapper.java

```diff
@@ -152,6 +141,47 @@ public SessionStateDescriptor getExistingIncompleteCaseDescriptor() {
         return null;
     }
 
+    /**
+     * @return
+     */
+    public static SessionStateDescriptor getFormStateForInterruptedUserSession() {
+        int idOfInterrupted = HiddenPreferences.getIdOfInterruptedSSD();
+        if (idOfInterrupted == -1) {
+            return null;
+        }
+        SqlStorage<SessionStateDescriptor> sessionStorage =
+                CommCareApplication.instance().getUserStorage(SessionStateDescriptor.class);
+        SessionStateDescriptor interrupted = sessionStorage.read(idOfInterrupted);
```

Comment:
Two quick things:

1)  I don't know _precisely_ why this could fail after fixing #2 below (no record exists), but I think that if this record isn't in session storage we shouldn't fail.
2) Shared preferences are per-app, not per-user, so right now this can bleed across userdbs (which could cause #1, but could also just cause bizarre behavior). Easiest fix is to append the user ID to the shared preferences key in the HiddenPreferences changes